examples/diff.c

```diff
@@ -25,6 +26,8 @@
  * how to do renaming detection and patch formatting.
  */
 
+char git_buf__initbuf[1];
```

Comment:
That's supposed to be a private implementation detail of `git_buf`, and there should be no need to define it — it's used as the default static value that git_bufs are initialized to point to. I believe things will break in strange ways if it gets duplicated.

Reply:
```
/usr/bin/ld: CMakeFiles/lg2.dir/diff.c.o: en la función `lg2_diff':
/home/alberto/projects/c/libgit2/examples/diff.c:116: referencia a `git_buf__initbuf' sin definir
collect2: error: ld devolvió el estado de salida 1
make[2]: *** [examples/CMakeFiles/lg2.dir/build.make:436: examples/lg2] Error 1
make[1]: *** [CMakeFiles/Makefile2:359: examples/CMakeFiles/lg2.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

I didn't find a way to solve this, maybe make diff depends on buffer.c in CMakeList.txt?

```
src/buffer.c
16:char git_buf__initbuf[1];
```

Reply:
Don't set `buf = git_buf__initbuf`, either set `buf = {0}` or don't initialize it and call `git_buf_init(&buf)`.

Reply:
ok, I was using `GIT_BUF_INIT` macro